vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,206 @@
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+import json
+import logging
+from builtins import set
+from datetime import datetime
+from datetime import timezone
+from io import BytesIO
+from typing import Iterable
+from zipfile import ZipFile
+
+import requests
+from packageurl import PackageURL
+from univers.version_range import PypiVersionRange
+from univers.versions import InvalidVersion
+from univers.versions import PypiVersion
+from univers.versions import SemverVersion
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import Importer
+from vulnerabilities.importer import Reference
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.severity_systems import SCORING_SYSTEMS
+
+logger = logging.getLogger(__name__)
+
+
+class PyPIImporter(Importer):
+    spdx_license_expression = "CC-BY-4.0"
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        """
+        1. Fetch the data from osv api
+        2. unzip the file
+        3. open the file one by one
+        4. yield the json file to parse_advisory_data
+        """
+        url = "https://osv-vulnerabilities.storage.googleapis.com/PyPI/all.zip"
+        try:
+            response = requests.get(url).content
+            with ZipFile(BytesIO(response)) as zip_file:
+                for file_name in zip_file.namelist():
+                    with zip_file.open(file_name) as f:
+                        vul_info = json.loads(f.read())
+                        yield parse_advisory_data(vul_info)
+        except requests.exceptions.RequestException:
+            logger.error("Failed to fetch osv-vulnerabilities PyPI")
+
+
+def parse_advisory_data(raw_data: dict) -> AdvisoryData:
+    summary = raw_data.get("summary") or ""
+    aliases = get_aliases(raw_data)
+    date_published = get_published_date(raw_data)
+    severity = get_severity(raw_data)
+    references = get_references(raw_data, severity)
+
+    affected_package = []
+    if "affected" in raw_data:
```

Comment:
Line 94:
```
   else:
        logger.error(f"affected_package not found - {raw_data['id'] !r}") 
```

Reply:
@ziadhany sorry for confusing you, I meant you can check inverse of that expression, this will save us from having a big code of block indented.
```
if not "affected" in raw_data:
    logger.error(f"affected_package not found - {raw_data['id'] !r}")
    # return the advisory with affected_package as an empty list 
---process affected and return advisory with affected_package from the data you got after processing ----
```

Reply:
> @ziadhany there is a field called "database_specific" in test data, do see if we can derive some severities from there

the problem is that “database_specific”,contains a JSON object with unspecified fields. 

npm:
```
 "database_specific": {
        "CWE": "CWE-327",
    }
```
but Python vulnerability :  
```
"database_specific": {
        "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2018/11/GHSA-2jc8-4r6g-282j/GHSA-2jc8-4r6g-282j.json"
      }
```